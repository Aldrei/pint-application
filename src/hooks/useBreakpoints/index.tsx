import { useTheme } from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';

export const useBreakpoints = () => {
  const theme = useTheme();

  const goSm = useMediaQuery(theme.breakpoints.up('sm'));
  const goMd = useMediaQuery(theme.breakpoints.up('md'));
  const goLg = useMediaQuery(theme.breakpoints.up('lg'));
  const goXl = useMediaQuery(theme.breakpoints.up('xl'));

  return [goSm, goMd, goLg, goXl];
};
