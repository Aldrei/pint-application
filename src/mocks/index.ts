export const PROPERTIES_LIST = {
  'paginate':{
    'data':[
      {
        'id':50,
        'client_id':1,
        'city_id':4303905,
        'neighborhood_id':2,
        'owner_id':46,
        'agent_id':1,
        'broker_id':1,
        'code':50,
        'codeTipo':null,
        'codePretty':null,
        'nomeImovel':'',
        'matricula':'',
        'lote':'',
        'quadra':'',
        'empreendimento':0,
        'placa':0,
        'possuiFoto':1,
        'exclusividade':0,
        'exclusividadePeriodoInicio':'',
        'exclusividadePeriodoFim':'',
        'finalidade':'',
        'categoria':'Residencial',
        'tipo':'Apartamento',
        'status':'Vendendo',
        'aluguelPeriodoInicio':'',
        'aluguelPeriodoFim':'',
        'dormitorio':'2',
        'garagem':'1',
        'lavanderia':1,
        'telhado':'',
        'forro':'',
        'piso':'',
        'aberturas':'',
        'alarme':1,
        'portaoEletronico':1,
        'pocoArtesiano':0,
        'cercaEletrica':0,
        'cameraDeVideo':0,
        'nascerDoSol':'',
        'descGeral':'02 dorm, sala,garagem.',
        'dataAgenciamento':'',
        'dataVenda':'',
        'apNomeCondominio':'',
        'apPredio':'',
        'apPavimento':'',
        'apTotalPavimentos':'',
        'apApto':'',
        'apElevador':1,
        'areaTotal':'',
        'areaConstruida':'',
        'areaFrente':'',
        'areaFundos':'',
        'areaDireita':'',
        'areaEsquerda':'',
        'valor':'',
        'valorCondicaoDeComissao':'',
        'valorPorcentagemDeComissao':'0.00',
        'valorPorcentagemDoCorretor':'0.00',
        'valorPorcentagemDoAgenciador':'0.00',
        'valorCondominio':'',
        'valorIPTUPago':'',
        'valorIPTU':'',
        'valorInssPago':'',
        'valorInss':'0.00',
        'valorIndiceDoCUB':'0.00',
        'valorAlgoParaRegularizar':'',
        'valorAlgoParaRegularizarDesc':'',
        'condAgente':'',
        'condPrestacao':'',
        'condSaldoDevedor':'',
        'condPrazo':'',
        'condReajuste':'',
        'condFGTS':0,
        'condDoacao':0,
        'condFinanciamento':0,
        'condCartaConsorcio':0,
        'condPartePermuta':0,
        'condSituacao':'',
        'condObs':'',
        'localEstado':'RS',
        'localIdCidade':'1',
        'localIdBairro':null,
        'localLogradouro':'',
        'localNumero':'',
        'localCEP':'',
        'localImediacoes':'',
        'latitude':null,
        'longitude':null,
        'zoom':null,
        'sitePublicarImovel':1,
        'sitePublicarValor':1,
        'siteImovelDestaque':0,
        'siteAcesso':'689',
        'sitePublicarMapa':null,
        'videoURL':'',
        'idUltimaTransacao':'0',
        'created_at':'2018-02-18 13:56:50',
        'updated_at':'2019-08-02 20:40:40',
        'desc':'Apartamento, 2 dormit\u00f3rios, garagem 1 carro',
        'title':'Apartamento, 2 dorm.',
        'hasExclusividade':false,
        'site_url':'http://csimovel.com/imovel/50/apartamento/campo-bom',
        'city':{
          'data':{
            'id':4303905,
            'name':'Campo Bom',
            'state':{
              'id':43,
              'name':'Rio Grande do Sul',
              'abbr':'RS',
              'created_at':'2018-02-18 16:56:49',
              'updated_at':'2018-02-18 16:56:49'
            },
            'long_desc':'Campo Bom, RS'
          }
        },
        'neighborhood':{
          'data':{
            'id':2,
            'client_id':1,
            'city_id':4303905,
            'idBairro':'',
            'nome':'Centro',
            'idCidade':'1',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:37',
            'city':{
              'data':{
                'id':4303905,
                'name':'Campo Bom',
                'state':{
                  'id':43,
                  'name':'Rio Grande do Sul',
                  'abbr':'RS',
                  'created_at':'2018-02-18 16:56:49',
                  'updated_at':'2018-02-18 16:56:49'
                },
                'long_desc':'Campo Bom, RS'
              }
            }
          }
        },
        'owner':{
          'data':{
            'id':46,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'idProprietario':'',
            'nomeRazao':'Olegario',
            'cpfCnpj':'',
            'rg':'',
            'cnh':'',
            'pessoa':'Fisica',
            'inscricaoEstadual':'',
            'sexo':'Masculino',
            'estadoCivil':'Casado(a)',
            'dataNascimento':'',
            'naturalidade':'',
            'profissao':'',
            'renda':'',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'fixo':'',
            'celular':'',
            'fax':'',
            'email':'',
            'email2':'',
            'obs':'',
            'conjNome':'',
            'conjCpf':'',
            'conjRg':'',
            'conjCnh':'',
            'conjDataNascimento':'0000-00-00',
            'conjNaturalidade':'',
            'conjProfissao':'',
            'conjRenda':'0.00',
            'conjPai':'',
            'conjMae':'',
            'conjMesmoEndereco':'o',
            'conjEstado':'',
            'conjIdCidade':'0',
            'conjIdBairro':'0',
            'conjLogradouro':'',
            'conjNumero':'',
            'conjCep':'',
            'conjApto':'',
            'conjFixo':'',
            'conjCelular':'',
            'conjFax':'',
            'conjEmail':'',
            'conjEmail2':'',
            'conjSpc':'',
            'conjSpcEntrada':'0000-00-00',
            'conjSpcSaida':'0000-00-00',
            'conjSpcValor':'0.00',
            'conjObs':'',
            'foto':'',
            'fotoMini':'',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:33'
          }
        },
        'broker':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'agent':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'photo':{
          'data':{
            'id':465,
            'property_id':50,
            'name':'5bf2583ec2633e60ecdb69ed247bc5ef.jpg',
            'ordem':1,
            'rotate':null,
            'thumb':'https://imobmobile.com.br/photos/thumb/5bf2583ec2633e60ecdb69ed247bc5ef.jpg',
            'normal':'https://imobmobile.com.br/photos/normal/5bf2583ec2633e60ecdb69ed247bc5ef.jpg'
          }
        }
      },
      {
        'id':56,
        'client_id':1,
        'city_id':4303905,
        'neighborhood_id':13,
        'owner_id':49,
        'agent_id':1,
        'broker_id':1,
        'code':56,
        'codeTipo':63,
        'codePretty':'CAA063',
        'nomeImovel':'',
        'matricula':'',
        'lote':'',
        'quadra':'',
        'empreendimento':0,
        'placa':0,
        'possuiFoto':1,
        'exclusividade':0,
        'exclusividadePeriodoInicio':'',
        'exclusividadePeriodoFim':'',
        'finalidade':'',
        'categoria':'Residencial',
        'tipo':'Casa Alvenaria',
        'status':'Vendendo',
        'aluguelPeriodoInicio':'',
        'aluguelPeriodoFim':'',
        'dormitorio':'3',
        'garagem':'3',
        'lavanderia':1,
        'telhado':'',
        'forro':'',
        'piso':'',
        'aberturas':'',
        'alarme':1,
        'portaoEletronico':1,
        'pocoArtesiano':0,
        'cercaEletrica':1,
        'cameraDeVideo':1,
        'nascerDoSol':'',
        'descGeral':'03 dorm (1 su\u00edte c/hidro e closet ),cozinha sob medida,pain\u00e9is,porcelanato,gesso,\u00e1gua quente,lareira,piscina,persianas automatizadas.',
        'dataAgenciamento':'',
        'dataVenda':'',
        'apNomeCondominio':'',
        'apPredio':'',
        'apPavimento':'',
        'apTotalPavimentos':'',
        'apApto':'',
        'apElevador':0,
        'areaTotal':'300.00',
        'areaConstruida':'229.00',
        'areaFrente':'12.00',
        'areaFundos':'12.00',
        'areaDireita':'25.00',
        'areaEsquerda':'25.00',
        'valor':'850000.00',
        'valorCondicaoDeComissao':'',
        'valorPorcentagemDeComissao':'0.00',
        'valorPorcentagemDoCorretor':'0.00',
        'valorPorcentagemDoAgenciador':'0.00',
        'valorCondominio':'',
        'valorIPTUPago':'',
        'valorIPTU':'',
        'valorInssPago':'',
        'valorInss':'0.00',
        'valorIndiceDoCUB':'0.00',
        'valorAlgoParaRegularizar':'',
        'valorAlgoParaRegularizarDesc':'',
        'condAgente':'',
        'condPrestacao':'',
        'condSaldoDevedor':'',
        'condPrazo':'',
        'condReajuste':'',
        'condFGTS':0,
        'condDoacao':0,
        'condFinanciamento':0,
        'condCartaConsorcio':0,
        'condPartePermuta':0,
        'condSituacao':'',
        'condObs':'',
        'localEstado':'RS',
        'localIdCidade':'1',
        'localIdBairro':null,
        'localLogradouro':'Rodrigo Jos\u00e9 Piltz',
        'localNumero':'83',
        'localCEP':'',
        'localImediacoes':'',
        'latitude':null,
        'longitude':null,
        'zoom':null,
        'sitePublicarImovel':0,
        'sitePublicarValor':0,
        'siteImovelDestaque':0,
        'siteAcesso':'1704',
        'sitePublicarMapa':null,
        'videoURL':'',
        'idUltimaTransacao':'0',
        'created_at':'2018-02-18 13:56:50',
        'updated_at':'2021-08-11 14:04:05',
        'valorPub':'850.000,00',
        'desc':'Casa Alvenaria, 3 dormit\u00f3rios, garagem 3 carros',
        'title':'Casa Alvenaria, 3 dorm.',
        'hasExclusividade':false,
        'site_url':'http://csimovel.com/imovel/56/casa-alvenaria/campo-bom',
        'city':{
          'data':{
            'id':4303905,
            'name':'Campo Bom',
            'state':{
              'id':43,
              'name':'Rio Grande do Sul',
              'abbr':'RS',
              'created_at':'2018-02-18 16:56:49',
              'updated_at':'2018-02-18 16:56:49'
            },
            'long_desc':'Campo Bom, RS'
          }
        },
        'neighborhood':{
          'data':{
            'id':13,
            'client_id':1,
            'city_id':4303905,
            'idBairro':'',
            'nome':'Solar do Campo',
            'idCidade':'1',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:37',
            'city':{
              'data':{
                'id':4303905,
                'name':'Campo Bom',
                'state':{
                  'id':43,
                  'name':'Rio Grande do Sul',
                  'abbr':'RS',
                  'created_at':'2018-02-18 16:56:49',
                  'updated_at':'2018-02-18 16:56:49'
                },
                'long_desc':'Campo Bom, RS'
              }
            }
          }
        },
        'owner':{
          'data':{
            'id':49,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'idProprietario':'',
            'nomeRazao':'Eliezer',
            'cpfCnpj':'',
            'rg':'',
            'cnh':'',
            'pessoa':'Fisica',
            'inscricaoEstadual':'',
            'sexo':'Masculino',
            'estadoCivil':'Casado(a)',
            'dataNascimento':'',
            'naturalidade':'',
            'profissao':'',
            'renda':'',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'fixo':'',
            'celular':'(51) 8236-4900',
            'fax':'',
            'email':'',
            'email2':'',
            'obs':'',
            'conjNome':'',
            'conjCpf':'',
            'conjRg':'',
            'conjCnh':'',
            'conjDataNascimento':'0000-00-00',
            'conjNaturalidade':'',
            'conjProfissao':'',
            'conjRenda':'0.00',
            'conjPai':'',
            'conjMae':'',
            'conjMesmoEndereco':'1',
            'conjEstado':'',
            'conjIdCidade':'0',
            'conjIdBairro':'0',
            'conjLogradouro':'',
            'conjNumero':'',
            'conjCep':'',
            'conjApto':'',
            'conjFixo':'(51) 9388-0596',
            'conjCelular':'(51) 9236-4900',
            'conjFax':'',
            'conjEmail':'',
            'conjEmail2':'',
            'conjSpc':'',
            'conjSpcEntrada':'0000-00-00',
            'conjSpcSaida':'0000-00-00',
            'conjSpcValor':'0.00',
            'conjObs':'',
            'foto':'',
            'fotoMini':'',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:33'
          }
        },
        'broker':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'agent':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'photo':{
          'data':{
            'id':491,
            'property_id':56,
            'name':'271ccaf01e123f96c9174dde8a65af5d.jpg',
            'ordem':1,
            'rotate':null,
            'thumb':'https://imobmobile.com.br/photos/thumb/271ccaf01e123f96c9174dde8a65af5d.jpg',
            'normal':'https://imobmobile.com.br/photos/normal/271ccaf01e123f96c9174dde8a65af5d.jpg'
          }
        }
      },
      {
        'id':63,
        'client_id':1,
        'city_id':4303905,
        'neighborhood_id':6,
        'owner_id':56,
        'agent_id':1,
        'broker_id':1,
        'code': null,
        'codeTipo':null,
        'codePretty':null,
        'nomeImovel':'',
        'matricula':'',
        'lote':'',
        'quadra':'',
        'empreendimento':0,
        'placa':1,
        'possuiFoto':1,
        'exclusividade':0,
        'exclusividadePeriodoInicio':'',
        'exclusividadePeriodoFim':'',
        'finalidade':'',
        'categoria':'Residencial',
        'tipo':'Terreno',
        'status': '',
        'aluguelPeriodoInicio':'',
        'aluguelPeriodoFim':'',
        'dormitorio':'0',
        'garagem':'0',
        'lavanderia':0,
        'telhado':'',
        'forro':'',
        'piso':'',
        'aberturas':'',
        'alarme':0,
        'portaoEletronico':0,
        'pocoArtesiano':0,
        'cercaEletrica':0,
        'cameraDeVideo':0,
        'nascerDoSol':'',
        'descGeral':'Terreno com linda vista.',
        'dataAgenciamento':'',
        'dataVenda':'',
        'apNomeCondominio':'',
        'apPredio':'',
        'apPavimento':'',
        'apTotalPavimentos':'',
        'apApto':'',
        'apElevador':0,
        'areaTotal':'310.15',
        'areaConstruida':'',
        'areaFrente':'10.65',
        'areaFundos':'10.74',
        'areaDireita':'29.00',
        'areaEsquerda':'29.00',
        'valor':'84800.00',
        'valorCondicaoDeComissao':'',
        'valorPorcentagemDeComissao':'0.00',
        'valorPorcentagemDoCorretor':'0.00',
        'valorPorcentagemDoAgenciador':'0.00',
        'valorCondominio':'',
        'valorIPTUPago':'',
        'valorIPTU':'',
        'valorInssPago':'',
        'valorInss':'0.00',
        'valorIndiceDoCUB':'0.00',
        'valorAlgoParaRegularizar':'',
        'valorAlgoParaRegularizarDesc':'',
        'condAgente':'',
        'condPrestacao':'',
        'condSaldoDevedor':'',
        'condPrazo':'',
        'condReajuste':'',
        'condFGTS':0,
        'condDoacao':0,
        'condFinanciamento':0,
        'condCartaConsorcio':0,
        'condPartePermuta':0,
        'condSituacao':'',
        'condObs':'',
        'localEstado':'RS',
        'localIdCidade':'1',
        'localIdBairro':null,
        'localLogradouro':'Waldemar Mossmann',
        'localNumero':'',
        'localCEP':'',
        'localImediacoes':'',
        'latitude':null,
        'longitude':null,
        'zoom':null,
        'sitePublicarImovel':0,
        'sitePublicarValor':0,
        'siteImovelDestaque':1,
        'siteAcesso':'34',
        'sitePublicarMapa':null,
        'videoURL':'',
        'idUltimaTransacao':'0',
        'created_at':'2018-02-18 13:56:50',
        'updated_at':'2020-02-25 18:23:49',
        'valorPub':'84.800,00',
        'desc':'Terreno',
        'title':'Terreno',
        'hasExclusividade':false,
        'site_url':'http://csimovel.com/imovel/63/terreno/campo-bom',
        'city':{
          'data':{
            'id':4303905,
            'name':'Campo Bom',
            'state':{
              'id':43,
              'name':'Rio Grande do Sul',
              'abbr':'RS',
              'created_at':'2018-02-18 16:56:49',
              'updated_at':'2018-02-18 16:56:49'
            },
            'long_desc':'Campo Bom, RS'
          }
        },
        'neighborhood':{
          'data':{
            'id':6,
            'client_id':1,
            'city_id':4303905,
            'idBairro':'',
            'nome':'Morada da Brisa',
            'idCidade':'1',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:37',
            'city':{
              'data':{
                'id':4303905,
                'name':'Campo Bom',
                'state':{
                  'id':43,
                  'name':'Rio Grande do Sul',
                  'abbr':'RS',
                  'created_at':'2018-02-18 16:56:49',
                  'updated_at':'2018-02-18 16:56:49'
                },
                'long_desc':'Campo Bom, RS'
              }
            }
          }
        },
        'owner':{
          'data':{
            'id':56,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'idProprietario':'',
            'nomeRazao':'Luiz Meister',
            'cpfCnpj':'',
            'rg':'',
            'cnh':'',
            'pessoa':'Fisica',
            'inscricaoEstadual':'',
            'sexo':'Masculino',
            'estadoCivil':'Casado(a)',
            'dataNascimento':'',
            'naturalidade':'',
            'profissao':'',
            'renda':'',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'fixo':'',
            'celular':'(51) 9288-5199',
            'fax':'',
            'email':'',
            'email2':'',
            'obs':'',
            'conjNome':'',
            'conjCpf':'',
            'conjRg':'',
            'conjCnh':'',
            'conjDataNascimento':'0000-00-00',
            'conjNaturalidade':'',
            'conjProfissao':'',
            'conjRenda':'0.00',
            'conjPai':'',
            'conjMae':'',
            'conjMesmoEndereco':'1',
            'conjEstado':'',
            'conjIdCidade':'0',
            'conjIdBairro':'0',
            'conjLogradouro':'',
            'conjNumero':'',
            'conjCep':'',
            'conjApto':'',
            'conjFixo':'',
            'conjCelular':'',
            'conjFax':'',
            'conjEmail':'',
            'conjEmail2':'',
            'conjSpc':'',
            'conjSpcEntrada':'0000-00-00',
            'conjSpcSaida':'0000-00-00',
            'conjSpcValor':'0.00',
            'conjObs':'',
            'foto':'',
            'fotoMini':'',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:33'
          }
        },
        'broker':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'agent':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'photo': null
      },
    ],
    'meta':{
      'pagination':{
        'total':272,
        'count':10,
        'per_page':10,
        'current_page':2,
        'total_pages':28,
        'links':{
          'previous':'https://imobmobile.com.br/api/properties?page=1',
          'next':'https://imobmobile.com.br/api/properties?page=3'
        }
      }
    }
  }
};
