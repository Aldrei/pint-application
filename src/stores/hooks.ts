import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import type { RootState, AppDispatch } from './index';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const useSelectorCallback = (state: RootState, reducer: keyof RootState) => state[reducer];
