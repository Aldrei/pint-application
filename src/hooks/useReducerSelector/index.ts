import { RootState, IReducersType } from '../../store';
import { useAppSelector } from '../../store/hooks';

export const useAppSelectorBlaBlaBal = (reducer: keyof IReducersType) => useAppSelector((state: RootState) => state[reducer]);