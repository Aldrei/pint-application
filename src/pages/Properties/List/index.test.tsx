import React from 'react';

import { render } from '@testing-library/react';

import { useLocation } from 'react-router-dom';
import { 
  useAppSelector, 
  useAppDispatch 
} from '../../../store/hooks';
import { useAppSelectorBlaBlaBal } from '../../../hooks/useReducerSelector';

import { PROPERTIES_LIST } from '../../../mocks';

import PropertiesList from './index';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
  useLocation: jest.fn(),
}));

jest.mock('../../../store/hooks', () => ({
  useAppSelector: jest.fn(),
  useAppDispatch: jest.fn(),
}));

jest.mock('../../../hooks/useReducerSelector', () => ({
  useAppSelectorBlaBlaBal: jest.fn()
}));

describe('Properties list page', () => {
  const useLocationMocked = useLocation as jest.MockedFunction<typeof useLocation>;
  const useAppSelectorMocked = useAppSelector as jest.MockedFunction<typeof useAppSelector>;
  const useAppDispatchMocked = useAppDispatch as jest.MockedFunction<typeof useAppDispatch>;
  const useAppSelectorBlaBlaBalMocked = useAppSelectorBlaBlaBal as jest.MockedFunction<typeof useAppSelectorBlaBlaBal>;

  beforeEach(() => {
    jest.resetAllMocks();
    useAppSelectorBlaBlaBalMocked.mockReturnValue({ status: 'success' });
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('Should be render data list', () => {
    useAppSelectorMocked.mockReturnValue({ data: PROPERTIES_LIST });

    useLocationMocked.mockReturnValue({
      search: '',
      state: undefined,
      key: '',
      pathname: '',
      hash: ''
    });

    useAppDispatchMocked.mockReturnValue(jest.fn());

    const nodeEl = render(<PropertiesList />);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  it('Should be render data list without infos', () => {
    useAppSelectorMocked.mockReturnValue({ data: PROPERTIES_LIST });

    useLocationMocked.mockReturnValue({
      search: '',
      state: undefined,
      key: '',
      pathname: '',
      hash: ''
    });

    useAppDispatchMocked.mockReturnValue(jest.fn());

    const nodeEl = render(<PropertiesList />);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  it('Should be render empty data list', () => {
    useAppSelectorMocked.mockReturnValue({ data: null });

    useLocationMocked.mockReturnValue({
      search: '',
      state: undefined,
      key: '',
      pathname: '',
      hash: ''
    });

    useAppDispatchMocked.mockReturnValue(jest.fn());

    const nodeEl = render(<PropertiesList />);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  it('Should be render loading', () => {
    useAppSelectorMocked.mockReturnValue({ data: null });

    useLocationMocked.mockReturnValue({
      search: '',
      state: undefined,
      key: '',
      pathname: '',
      hash: ''
    });

    useAppDispatchMocked.mockReturnValue(jest.fn());

    useAppSelectorBlaBlaBalMocked.mockReturnValue({ status: 'loading' });

    const nodeEl = render(<PropertiesList />);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });
});