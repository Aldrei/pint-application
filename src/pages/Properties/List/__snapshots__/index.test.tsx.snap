// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties list page Should be render data list 1`] = `
<body>
  <div>
    <div
      class="sc-bczRLJ khWAxr"
      data-testid="propertiesList-container"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="css-a6j84v"
        >
          <div
            class="css-1iz4iln"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-pwxpp9-MuiInputBase-root"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              placeholder="Pesquisar..."
              type="text"
              value=""
            />
          </div>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary Mui-focused css-8yhqw3-MuiInputBase-root"
          >
            <input
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              type="text"
              value=""
            />
          </div>
          <div
            class="MuiBox-root css-0"
          />
        </div>
      </div>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        style="width: 100%;"
      />
      <div
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        role="separator"
        style="margin: 20px 20px 30px;"
      />
      <div
        class="css-1p5q5e5-MuiStack-root"
      >
        <nav
          aria-label="pagination navigation"
          class="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
        >
          <ul
            class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeLarge MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-previousNext css-1ysyrvn-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-n8417t-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateBeforeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeLarge MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-previousNext css-1ysyrvn-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-n8417t-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateNextIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</body>
`;
